### PR TITLE
feat: support date-fns in Picker

### PR DESCRIPTION
**Pull request by fireairforce**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
ref issue is: 
https://github.com/ant-design/ant-design/issues/22732
https://github.com/ant-design/ant-design/issues/21866

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Support date-fns library in Picker.      |
| 🇨🇳 Chinese |    DatePicker 组件支持 date-fns 日期库。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered docs/react/replace-moment.en-US.md](https://github.com/fireairforce/ant-design/blob/date-fns/docs/react/replace-moment.en-US.md)
[View rendered docs/react/replace-moment.zh-CN.md](https://github.com/fireairforce/ant-design/blob/date-fns/docs/react/replace-moment.zh-CN.md)

### CODE REVIEW

File: docs/react/replace-moment.zh-CN.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: 替换 Moment.js
 
 ## 如何在 DatePicker 中使用自定义日期库（如 dayjs ）？
 
-考虑到包的大小，你可以用自定义日期库替换 Moment。在这里我们提供了两种方式来实现替换:
+考虑到包的大小，你可以用自定义日期库(目前支持 dayjs、date-fns)替换 Moment。在这里我们提供了两种方式来实现替换:
```
Comment (07akioni) [inline docs/react/replace-moment.zh-CN.md:8]: 替换方式能类比么，从 datefns 导出的东西和 dayjs 一致？还有对 date-fns 的支持要加上版本提醒

Comment (07akioni) [inline docs/react/replace-moment.zh-CN.md:8]: 替换方式能类比么，从 datefns 导出的东西和 dayjs 一致？还有对 date-fns 的支持要加上版本提醒

Comment (fireairforce) [reply docs/react/replace-moment.zh-CN.md:8]: 嗯嗯，是类似的，都是通过 generate 的方式去生成，这里有些细节我补一下。

**Comment by fireairforce** [comment]
新增了一下文档~可以CC

File: docs/react/replace-moment.en-US.md [hunk 2]
```diff
@@ -104,3 +104,25 @@ If you use [umi](https://umijs.org/), you can reference [antd4-use-dayjs-replace
...
+
+## Use date-fns
+
+`date-fns` currently supports custom component methods similar to `dayjs`. The difference is that the parameter types used are different. Support is provided in antd 4.5.0 and above.
+
+For Example:
+
+### DatePicker.tsx
+
+Create `src/components/DatePicker.tsx`。
```
Comment (07akioni) [inline docs/react/replace-moment.en-US.md:116]: 英文句号

File: docs/react/replace-moment.en-US.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ title: Replace Moment.js
 
 ## How to use DatePicker with a custom date library like dayjs
 
-Considering it's bundle size, you might want to replace Moment.js with a different date library of your choice. We provide two ways to customize which date library is used:
+Considering it's bundle size, you might want to replace Moment.js with a different date library(now support dayjs and date-fns) of your choice. We provide two ways to customize which date library is used:
```
Comment (xrkffgg) [inline docs/react/replace-moment.en-US.md:8]:
```suggestion
Considering it's bundle size, you might want to replace Moment.js with a different date library (now support dayjs and date-fns) of your choice. We provide two ways to customize which date library is used:
```

**Comment by fireairforce** [comment]
已经修改了~

**Comment by fireairforce** [comment]
image

lint看上去不是我这边的，我这边已经 rebase 到最新的了～

**Comment by Kermit-Xuan** [comment]
@fireairforce  重新rebase一下，然后把 package.json 里面的版本改动删了，主分支上已经更新到 `1.15.0` 了。

**Comment by Kermit-Xuan** [comment]
@fireairforce 这个对应的依赖已经在主分支上更新了，`rc-picker@1.15.0`

**Comment by Kermit-Xuan** [comment]
@afc163 这个直接合到master吧？

**Comment by fireairforce** [comment]
我用feature rebase 了一次

**Comment by fireairforce** [comment]
现在应该阔以了～

**Review by 07akioni — APPROVED** [review]

**Comment by paxarpp** [comment]
hello, the suggested use case for date-fns does not work for ru localization,
perhaps there is some solution for localization other than the default en

**Comment by Makentosh** [comment]
Hello,
yes,really date-fns does not work if localization through configprovider is ru
for a long time I couldn't understand why the datepicker doesn't work

**Pull request merged**